### ISSUE TITLE
Checker should have option to check code comments and strings only.

### ISSUE THREAD

**Issue by justin-romano** [Jason-Rev/vscode-spell-checker#150]
Not all symbols in code are real words. a lot are abbreviations and pseudo words
I only want strings and comments to be spell checked

**Comment by Jason3S** [comment]
This is very similar to #116. It is something I want to do.

Which programming language are you using?

I intend to add a setting. But each programming language has its own definitions of a comment or string. VS Code does not expose language grammar to the extensions. Simple regular expressions work most of the time, but it is very easy to fool them. Which means I would need to implement a full on equivalent to the colorizer.

**Comment by justin-romano** [comment]
I would have thought that you could identify comments and strings from the syntax engine for all supported languages in vscode.
I'm using TS btw.

**Comment by Jason3S** [comment]
Yes, me too. But the syntax highlighting interface is not exposed by VS Code.

**Comment by justin-romano** [comment]
"If it walks like a duck and talks like a duck, it’s a duck, right? So if this duck is not giving you the noise that you want, you’ve got to just punch that duck until it returns what you expect"

**Comment by justin-romano** [comment]
punch that duck @Jason3S :D

**Comment by danielcrk** [comment]
I would really like to see this feature as well!

**Comment by Ace-Cassidy** [comment]
Agreed, in fact i'm having trouble at the moment where spellcheck is happening on python source but NOT on comments. Don't know why

**Comment by Jason3S** [comment]
My plan is to make it a selectable choice in the UI. It will take a bit of time to work on it.

**Comment by Jason3S** [comment]
@Ace-Cassidy please open a new issue and any steps to reproduce. Include `cspell.json` or `.vscode/settings.json`.
